### PR TITLE
 Add enabled integration test for vulnerability detector

### LINKED ISSUE #668: Add enabled to vulnerability detector integration tests
# Vulnerability detector

| Version | Revision |
| --- | --- |
| 3.12.3| 31208 |

## Summary
- [ ] Test enabled `yes`
- [ ] Test enabled `no`

## Test enabled parameter

**Short description**

Vulnerability-detector must be active or not on start depending on the value of the `enabled` parameter, and generate the correct logs.

**Category**

Vulnerability-detector

**Subcategory**

General settings

**Configuration sample**

``` XML
  <vulnerability-detector>
    <enabled>yes</enabled>
    <interval>100d</interval>
    <run_on_start>yes</run_on_start>
    <provider name="canonical">
      <enabled>yes</enabled>
      <os update_interval="10m">precise</os>
      <os update_interval="20m">trusty</os>
      <os update_interval="30m">xenial</os>
      <os>bionic</os>
      <update_interval>40m</update_interval>
    </provider>
  </vulnerability-detector>
```

**Expected logs**
```
2020/04/21 08:10:39 wazuh-modulesd:vulnerability-detector[71109] wm_vuln_detector.c:5238 at wm_vuldet_init(): DEBUG: Module disabled. Exiting...
```
Regards,
Daniel Folch

### PR DESCRIPTION
**Pull request by DFolchA**
Hello team,

This PR closes #668 and adds the test for the `enabled` parameter to  the vulnerability detector tests

To perform the test we look for the corresponding debug logs in `ossec.log`.

### Tests logic

When enabled is set to `no` the following log will appear:

```
2020/04/21 08:10:39 wazuh-modulesd:vulnerability-detector[71109] wm_vuln_detector.c:5238 at wm_vuldet_init(): DEBUG: Module disabled. Exiting...
```

On the other hand, if enabled is set to 'yes' then that log will not appear.

### Tests checks

If everything goes as expected:

```
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.8', 'Platform': 'Linux-3.10.0-957.12.2.el7.x86_64-x86_64-with-centos-7.6.1810-Core', 'Packages': {'pytest': '5.4.1', 'py': '1.8.1', 'pluggy': '0.13.1'}, 'Plugins': {'metadata': '1.8.0', 'html': '2.0.1', 'testinfra': '5.0.0'}}
rootdir: /home/vagrant/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1, testinfra-5.0.0
collected 4 items                                                                                      

test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration0-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled] PASSED [ 25%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration0-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled] SKIPPED [ 50%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration1-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled] SKIPPED [ 75%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration1-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled] PASSED [100%]

===================================== 2 passed, 2 skipped in 3.47s =====================================
```

In case the messages do not appear in the log:

```
========================================= test session starts ==========================================
platform linux -- Python 3.6.8, pytest-5.4.1, py-1.8.1, pluggy-0.13.1 -- /bin/python3
cachedir: .pytest_cache
metadata: {'Python': '3.6.8', 'Platform': 'Linux-3.10.0-957.12.2.el7.x86_64-x86_64-with-centos-7.6.1810-Core', 'Packages': {'pytest': '5.4.1', 'py': '1.8.1', 'pluggy': '0.13.1'}, 'Plugins': {'metadata': '1.8.0', 'html': '2.0.1', 'testinfra': '5.0.0'}}
rootdir: /home/vagrant/wazuh-qa/tests/integration, inifile: pytest.ini
plugins: metadata-1.8.0, html-2.0.1, testinfra-5.0.0
collected 4 items                                                                                      

test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration0-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled] FAILED [ 25%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration0-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled] SKIPPED [ 50%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration1-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled] SKIPPED [ 75%]
test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration1-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled] FAILED [100%]

=============================================== FAILURES ===============================================
_ test_enabled[get_configuration0-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled] _
test_vulnerability_detector/test_enabled/test_enabled.py:64: in test_enabled
    error_message=custom_error_message)
/usr/local/lib/python3.6/site-packages/wazuh_testing/tools/monitoring.py:187: in start
    error_message=error_message).result()
/usr/local/lib/python3.6/site-packages/wazuh_testing/tools/monitoring.py:405: in start
    raise TimeoutError()
E   TimeoutError
----------------------------------------- Captured stderr call -----------------------------------------
2020-05-05 10:14:29,876 - wazuh_testing - ERROR - Vulnerability detector is disabled
2020-05-05 10:14:29,877 - wazuh_testing - ERROR - Results accumulated: 0
2020-05-05 10:14:29,877 - wazuh_testing - ERROR - Results expected: 1
------------------------------------------ Captured log call -------------------------------------------
ERROR    wazuh_testing:monitoring.py:401 Vulnerability detector is disabled
ERROR    wazuh_testing:monitoring.py:402 Results accumulated: 0
ERROR    wazuh_testing:monitoring.py:404 Results expected: 1
_ test_enabled[get_configuration1-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled] _
test_vulnerability_detector/test_enabled/test_enabled.py:64: in test_enabled
    error_message=custom_error_message)
/usr/local/lib/python3.6/site-packages/wazuh_testing/tools/monitoring.py:187: in start
    error_message=error_message).result()
/usr/local/lib/python3.6/site-packages/wazuh_testing/tools/monitoring.py:405: in start
    raise TimeoutError()
E   TimeoutError
----------------------------------------- Captured stderr call -----------------------------------------
2020-05-05 10:14:44,483 - wazuh_testing - ERROR - Vulnerability detector is enabled
2020-05-05 10:14:44,483 - wazuh_testing - ERROR - Results accumulated: 0
2020-05-05 10:14:44,483 - wazuh_testing - ERROR - Results expected: 1
------------------------------------------ Captured log call -------------------------------------------
ERROR    wazuh_testing:monitoring.py:401 Vulnerability detector is enabled
ERROR    wazuh_testing:monitoring.py:402 Results accumulated: 0
ERROR    wazuh_testing:monitoring.py:404 Results expected: 1
--------------------------------------- Captured stdout teardown ---------------------------------------
2020-05-05T10:14:44.1588673684
======================================= short test summary info ========================================
FAILED test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration0-tags_to_apply0-callback_detect_vulnerability_detector_enabled-Vulnerability detector is disabled]
FAILED test_vulnerability_detector/test_enabled/test_enabled.py::test_enabled[get_configuration1-tags_to_apply1-callback_detect_vulnerability_detector_disabled-Vulnerability detector is enabled]
==================================== 2 failed, 2 skipped in 29.68s =====================================
```

Regards,
Daniel Folch

### CODE REVIEW

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py [hunk 1]
```diff
@@ -0,0 +1,19 @@
+# Copyright (C) 2015-2020, Wazuh Inc.
+# Created by Wazuh, Inc. <info@wazuh.com>.
+# This program is free software; you can redistribute it and/or modify it under the terms of GPLv2
+
+import re
+
+def callback_detect_vulnerability_detector_disabled(line)
```
Comment (BraulioV) [inline deps/wazuh_testing/wazuh_testing/vulnerability_detector.py:7]: `:` is missing here.

File: tests/integration/test_vulnerability_detector/test_enabled/test_enabled.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+def test_enabled(tags_to_apply, custom_callback, custom_error_message, 
+                      get_configuration, configure_environment,
+                      restart_modulesd):
+    """
+    Check if is enabled when the enabled parameter is set to 'yes'
+    ----------
+    custom_callback : function
+        Callback that will be applied to the value of tags to apply.
+    custom_error_message : str
+        Custom error message that will be shown if the test fails.
```
Comment (BraulioV) [inline tests/integration/test_vulnerability_detector/test_enabled/test_enabled.py:61]: Remove these parameters from the docstring.

File: deps/wazuh_testing/wazuh_testing/vulnerability_detector.py (hunk 1 above)
Comment (DFolchA) [reply deps/wazuh_testing/wazuh_testing/vulnerability_detector.py]: Done in 2d7be42.

File: tests/integration/test_vulnerability_detector/test_enabled/test_enabled.py (hunk 2 above)
Comment (DFolchA) [reply tests/integration/test_vulnerability_detector/test_enabled/test_enabled.py]: Done in 2d7be42.

**Pull request merged**